src/Models/ShallowWaterModels/shallow_water_advection_operators.jl

```diff
@@ -9,13 +10,28 @@ function div_hUv(i, j, k, grid, advection, solution)
 end
 
 @inline momentum_flux_huu(i, j, k, grid, advection, solution) =
-    @inbounds momentum_flux_uu(i, j, k, grid, solution.uh, solution.uh) / h[i, j, k]
+    @inbounds momentum_flux_uu(i, j, k, grid, advection, solution.uh, solution.uh) / solution.h[i, j, k]
 
 @inline momentum_flux_huv(i, j, k, grid, advection, solution) =
-    @inbounds momentum_flux_uv(i, j, k, grid, solution.uh, solution.vh) / ℑxyᶠᶠᵃ(i, j, k, grid, h)
+    @inbounds momentum_flux_uv(i, j, k, grid, advection, solution.uh, solution.vh) / ℑxyᶠᶠᵃ(i, j, k, grid, solution.h)
```

Comment:
Do we need to use high-order interpolation for `h` here? It might be inconsistent (or just effectively second-order) if we use high-order intepolation for `uh` and `vh` but then just second-order interpolation for `h`.

Two-way high-order intepolation for `h` is slightly annoying, but doable. You'll need to use functions like `_symmetric_interpolate_yᵃᶠᵃ`, etc:

https://github.com/CliMA/Oceananigans.jl/blob/master/src/Advection/centered_advective_fluxes.jl#L15

As a side note, its a bit messy (but makes sense) that we have special interpolation functions dispatching on the _advection_ scheme. High-order interpolation isn't really specific to advection; it's just that we only use it for advection currently. A cleaner design would separate order of intepolation from the concept of advection, and perhaps provide generic interpolation and differentiation operators for n-th order (like what we have for second-order interpolation and differentiation). Food for thought.

Reply:
Could make sense to save fixing `h` interpolation for a future PR.

Reply:
At the moment we are just using 2nd order approximation to the gradient of pressure, so using a 2nd order interolation for h is fine.  However, if we want higher order approximations for pressure, then this is something that needs to be sorted out.  This is not conceptually difficult but it will need to be tested and confirmed that it's worth the extra effort.  But that will be for another PR.

Short answer, yes.